### ISSUE TITLE
marian-decoder seems to miss calculate the best translation

### ISSUE THREAD

**Issue by adjouama** [marian-nmt/marian#314]
Hello,

I trained a transformer with SentencePiece (En-Fr). 
I reached 61 blue-detoke score. When decoding using `--n-best` parameter, the best translation is not ranked with the highest score (see log below):

`[2020-02-13 11:36:13] Best translation 0 : The Access and Transport Architecture Work
[2020-02-13 11:36:13] Best translation: 0 ||| The Access and Transport Architecture Work ||| F0= -1.38269 ||| -0.430195
0 ||| Travaux sur l'architecture d'accès et de transport ||| F0= -3.89615 ||| -0.877256
0 ||| Travaux relatifs à l'architecture d'accès et de transport ||| F0= -4.60489 ||| -0.988219
0 ||| Travaux sur l'architecture de l'accès et du transport ||| F0= -4.78057 ||| -1.02592
0 ||| Les travaux relatifs à l'architecture d'accès et de transport ||| F0= -5.27639 ||| -1.08308
0 ||| Travaux relatifs à l'architecture de l'accès et du transport ||| F0= -5.49288 ||| -1.12752
`

Basically, the sentence is not even translated (stays in the source language). 

Do you believe that this could be a model problem or the way marian-decoder scores translations ? 

Thank you a lot in advance

**Comment by emjotde** [comment]
Hi,
The log is telling you that it's a model problem. The score for the top translation is the highest, that's what the decoder is going to return. So it's not a decoder problem. If the best translation according to your or any human's opinion would always be the best results according to every model, search would not be a problem at all :) Unfortunately this is not the case. 

Most likely, you have quite a bit of untranslated sentences in your training data. Even if larger fragments are present that can happen.

**Comment by adjouama** [comment]
> Hi,
> The log is telling you that it's a model problem. The score for the top translation is the highest, that's what the decoder is going to return. So it's not a decoder problem. If the best translation according to your or any human's opinion would always be the best results according to every model, search would not be a problem at all :) Unfortunately this is not the case.
> 
> Most likely, you have quite a bit of untranslated sentences in your training data. Even if larger fragments are present that can happen.

That's right. I found some English segments in my target. 
Thank you a lot :-)
